### PR TITLE
Set Helm major version label for chart-operator app CRs

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/10798

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: pkg/label/app.go [hunk 1]
```diff
@@ -3,4 +3,7 @@ package label
 const (
 	// App is a standard label for tenant resources.
 	App = "app"
+
+	// AppOperatorHelmMajorVersion is a label for chart-operator app CRs.
+	AppOperatorHelmMajorVersion = "app-operator.giantswarm.io/helm-major-version"
```
Comment (rossf7) [inline pkg/label/app.go:8]: This is the new label. It controls which instance of app-operator bootstraps chart-operator based on the helm version.

File: service/controller/resource/app/desired.go [hunk 2]
```diff
@@ -58,7 +59,32 @@ func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) ([]*g8s
...
+					if err != nil {
+						return nil, microerror.Mask(err)
+					}
+
+					// For chart-operator 1.0.0 and greater we use Helm 3.
+					if version.Major() >= 1 {
+						helmMajorVersion = "3"
+					} else {
+						helmMajorVersion = "2"
+					}
```
Comment (tomahawk28) [inline service/controller/resource/app/desired.go:81]: I know sometimes we like to have fewer indentation blocks. 
```suggestion
                                            helmMajorVersion = "2"
					if version.Major() >= 1 {
						helmMajorVersion = "3"
					}
```

Comment (rossf7) [reply service/controller/resource/app/desired.go]: Yes sure. I tried to simplify it and removed the block as this is already in a conditional. 

PTAL

**Comment by tomahawk28** [comment]
LGTM :)

**Pull request merged**